docs/arch/adr-008-weave-to-flannel-migration.md

```diff
@@ -0,0 +1,39 @@
+# ADR 8: Weave to Flannel Migration
+
+## Context
+
+As proposed in ADR 6, the future CNI of kURL will be Flannel.
+However, almost all existing installations are running Weave.
+
+A migration path must be provided for those instances.
+
+There were two general proposals for a migration path.
+One was to use Multus-CNI to allow using both Flannel and Weave at the same time before stripping out Weave, allowing an upgrade while only taking down one node at once.
+The other was to run a migration that removed Weave from all instances simultaneously before installing Flannel.
+This would be far simpler, not require an additional IP range, and not require Multus to remain in the cluster going forwards.
+It would also require 'stopping the world' and cause downtime.
+
+## Decision
+
+Provide a stop-the-world migration path to move from Weave to Flannel.
+
+## Solution
+
+Create a script to do the following:
+
+1. synchronize execution on all cluster nodes to minimize downtime
+2. Uninstall weave, removing iptables rules and host files
+3. Edit kubeadm.conf on all primary nodes and run kubeadm init
+4. Install flannel
+5. Stop kubelet, flush iptables, restart containerd and start kubelet
+6. wait for nodes to be ready, and then recreate cluster-networking pods (kube-system, then CSI, then everything else)
+
```

Comment:
What we need to do to ensure that the migration was done successfully? 
Could we add some steps with basic checks?

Reply:
Yeah, there's already code in weave to ensure that inter-pod networking works, should be able to leverage that

Reply:
https://github.com/replicatedhq/kURL/blob/3876f5e7c24ab99d8680c45ec701a609e491f7c0/scripts/common/kubernetes.sh#L786

Incidentally I'm going to make a PR to make this have podantiaffinity, so that it'll run on different nodes when possible